GPU/Directx9/VertexShaderGeneratorDX9.cpp

```diff
@@ -279,6 +284,18 @@ void GenerateVertexShaderDX9(int prim, char *buffer, bool useHWTransform) {
 	}
 	WRITE(p, "};\n");
 
+	if (!gstate.isModeThrough()) {
+		// Apply the projection and viewport to get the Z buffer value, floor to integer, undo the viewport and projection.
+		// Not completely sure this is 100% right under DX9 as the Z range is different...
```

Comment:
Yes, that seems right, I was just too tired to figure it out. Will apply. Also it looks like we can easily do that by pre-adjusting the u_depthRange constant values.